dev/Repeater/APITests/Common/ReadOnlyNotifyPropertyChangedCollection.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Text;
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests.RepeaterTests.Common
+{
+    public class ReadOnlyNotifyPropertyChangedCollection<T> : IReadOnlyList<T>, INotifyCollectionChanged, IKeyIndexMapping
+    {
+        public ReadOnlyNotifyPropertyChangedCollection() { }
+
+        public ReadOnlyNotifyPropertyChangedCollection(IEnumerable<T> data)
+        {
+            Data = new ObservableCollection<T>(data);
+        }
+
+        #region IReadOnlyList<T>
+
+        public int Count => Data.Count;
+
+        public T this[int index] => Data[index];
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            return this.Data.GetEnumerator();
```

Comment:
> return this.Data.GetEnumerator(); [](start = 11, length = 34)

Hopefully with this change the Enumerator calls should not be made. Perhaps it would be better to throw an exception in these two calls to ensure we don't go down the path of enumerating the list.

Reply:
Good idea, done.